ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkWriter.java

```diff
@@ -1506,6 +1515,88 @@ private void writeGenerators(AmplExportContext context) throws IOException {
         }
     }
 
+    private void writeBatteries(AmplExportContext context) throws IOException {
+        try (Writer writer = new OutputStreamWriter(dataSource.newOutputStream("_network_batteries", "txt", append), StandardCharsets.UTF_8);
+             TableFormatter formatter = new AmplDatTableFormatter(writer,
+                     getTableTitle("Batteries"),
+                     AmplConstants.INVALID_FLOAT_VALUE,
+                     !append,
+                     AmplConstants.LOCALE,
+                     new Column(VARIANT),
+                     new Column("num"),
+                     new Column("bus"),
+                     new Column(CON_BUS),
+                     new Column(SUBSTATION),
+                     new Column(MINP),
+                     new Column(MAXP),
+                     new Column("p0 (MW)"),
+                     new Column("q0 (MW)"),
+                     new Column(MIN_Q_MAX_P),
+                     new Column(MIN_Q0),
+                     new Column(MIN_Q_MIN_P),
+                     new Column(MAX_Q_MAX_P),
+                     new Column(MAX_Q0),
+                     new Column(MAX_Q_MIN_P),
+                     new Column(FAULT),
+                     new Column(config.getActionType().getLabel()),
+                     new Column("id"),
+                     new Column(DESCRIPTION),
+                     new Column(ACTIVE_POWER),
+                     new Column(REACTIVE_POWER))) {
+            List<String> skipped = new ArrayList<>();
+            for (Battery b : network.getBatteries()) {
+                Terminal t = b.getTerminal();
+                Bus bus = AmplUtil.getBus(t);
+                int busNum = -1;
+                if (bus != null) {
+                    busNum = mapper.getInt(AmplSubset.BUS, bus.getId());
+                }
+                int conBusNum = -1;
+                // take connectable bus if exists
+                Bus conBus = AmplUtil.getConnectableBus(t);
+                if (conBus != null) {
+                    conBusNum = mapper.getInt(AmplSubset.BUS, conBus.getId());
+                }
+                // Always write Batteries
+
+                String id = b.getId();
+                context.batteryIdsToExport.add(id);
+                int num = mapper.getInt(AmplSubset.BATTERY, id);
+                int vlNum = mapper.getInt(AmplSubset.VOLTAGE_LEVEL, t.getVoltageLevel().getId());
+                double p0 = b.getP0();
+                double q0 = b.getQ0();
+                double minP = b.getMinP();
+                double maxP = b.getMaxP();
+
+                formatter.writeCell(variantIndex)
+                        .writeCell(num)
+                        .writeCell(busNum)
+                        .writeCell(conBusNum != -1 ? conBusNum : busNum)
+                        .writeCell(vlNum)
+                        .writeCell(minP)
+                        .writeCell(maxP)
+                        .writeCell(p0)
+                        .writeCell(q0)
```

Comment:
If you change the column order, don’t forget to adapt the code here

Reply:
Done